lib/omniauth/strategies/oidc/back_channel_logout_request.rb

```diff
@@ -50,18 +50,22 @@ def decode_logout_token(logout_token)
                 :verify_sub => true,
                 :leeway => 60
             }
-            if validate_logout_token(logout_jwt[0])
+            if validate_events(logout_jwt[0]) && validate_nonce(logout_jwt[0])
               @request.update_param('sid', logout_jwt[0]['sid'])
             else
               raise 'Logout JWT missing events claim or nonce claim is present'
             end
           end
         end
 
-        def validate_logout_token(logout_jwt)
+        def validate_events(logout_jwt)
           logout_jwt.key?('events') &&
-              logout_jwt['events'][0] == 'http://schemas.openid.net/event/backchannel-logout' &&
-              !logout_jwt.key?('nonce')
+              (logout_jwt['events'][0] == 'http://schemas.openid.net/event/backchannel-logout' ||
```

Comment:
Actually I tested it for both old and new versions and this check passes cause (the dictionary is empty for now)